File: meanCalculator.js
function mathFunc(data) {
    let mean = 0;
    for (let i = 0; i < data.length; i++) {
        mean+=data[i];
    }
    return mean/data.length;
}

function setup() {
    let data = [];
    let inp;
    while (true) {
        inp = prompt("enter a value ").toLowerCase();
        if (inp === "end") {
            console.log("selection finished");
            break;
        } else {
            if (isNaN(Number(inp))) {
                console.log("invalid response");
            } else {
                data.push(Number(inp));
            }
        }
    }
    console.log("mean = " + String(mathFunc(data)));
    while (inp !== "yes" && inp !== "no") {
        inp = prompt("do you wish to restart? ").toLowerCase();
        switch (inp) {
            case "yes":
                console.log("restarting...");
                setup();
                break;
            case "no":
                console.log("finished...");
                throw new Error();
                break;
            default:
                console.log("invalid response")
        }
    }
}
console.log("welcome!\ntype end to finish setup process");
setup();

File: meanCalculator.py
def mathFunc(data):
    mean = 0
    for item in data:
        mean+=item
    return mean/len(data)

def setup():
    data = []
    while True:
        inp = input("input a value ").lower()
        if inp == "end":
            print("selection finished")
            break
        else:
            try:
                data.append(float(inp))
            except:
                print("invalid response")
    print("mean = " + str(mathFunc(data)))
    while (True):
        inp = input("do you wish to restart? ").lower()
        if inp == "yes":
            print("restarting...")
            setup()
            break
        elif inp == "no":
            print("finishing...")
            exit()
        else:
            print("invalid response")
print("welcome!\ntype 'end' to finish data setup")
setup()
